src/index.ts

```diff
@@ -128,6 +129,60 @@ const getCustomScalarValue = (customScalar: ScalarDefinition, opts: Options<Name
     return value;
 };
 
+const getFakerGenerators = (generatorName: ScalarGeneratorName): [Function, string] => {
+    let embeddedGenerator: unknown = faker;
+    let dynamicGenerator = 'faker';
+
+    if (typeof generatorName === 'string') {
+        const generatorPath = generatorName.split('.');
+        for (const key of generatorPath) {
+            if (embeddedGenerator.hasOwnProperty(key)) {
+                embeddedGenerator = embeddedGenerator[key];
+                dynamicGenerator = `${dynamicGenerator}['${key}']`;
+            }
+        }
+    }
+
+    // If the faker generator is not a function, we can assume the path is wrong
+    if (typeof embeddedGenerator === 'function') {
+        return [embeddedGenerator, dynamicGenerator];
+    }
+
+    return [null, null];
+};
+
+const getFakerCustomScalarValue = (customScalar: ScalarDefinition, opts: Options<NamedTypeNode>) => {
+    // If there is a mapping to a `faker` type, then use it
+    const [embeddedGenerator, dynamicGenerator] = getFakerGenerators(customScalar.generator);
+    if (!embeddedGenerator && customScalar.generator) {
+        return customScalar.generator;
+    }
+
+    const generatorArgs: unknown[] = Array.isArray(customScalar.arguments)
+        ? customScalar.arguments
+        : [customScalar.arguments];
+    if (opts.dynamicValues) {
+        return `${dynamicGenerator}(...${JSON.stringify(generatorArgs)})`;
+    }
+    const value = embeddedGenerator(...generatorArgs);
+
+    if (typeof value === 'string') {
+        return `'${value}'`;
+    }
+    if (typeof value === 'object') {
+        return `${JSON.stringify(value)}`;
+    }
+    return value;
+};
+
+const getCustomScalarValue = (customScalar: ScalarDefinition, opts: Options<NamedTypeNode>) => {
+    if (opts.generateLibrary === 'casual') {
+        return getCasualCustomScalarValue(customScalar, opts);
+    }
+
+    return getFakerCustomScalarValue(customScalar, opts);
```

Comment:
You should probably check if the value is `faker` and throw an error if the value is unknown

Reply:
I had one, but typescript told me the value couldn't be unknown. I'll add it back!